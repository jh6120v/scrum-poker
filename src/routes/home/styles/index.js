import styled, { keyframes } from 'styled-components';
import { respondTo } from '../../../styles/_mixin';

const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-content: center;
  height: 100%;
  padding: 30px 15px;
  background: #e2e2e2;
  background: -moz-linear-gradient(45deg,  #e2e2e2 0%, #dbdbdb 50%, #d1d1d1 51%, #fefefe 100%);
  background: -webkit-linear-gradient(45deg,  #e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  background: linear-gradient(45deg,  #e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#e2e2e2', endColorstr='#fefefe',GradientType=1 );
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Logo = styled.div`
  width: calc((100% - 100px) / 2);
  height: 200px;
  background: url(${(props) => props.img}) no-repeat center;
  background-size: contain;
  animation: ${rotate} 3s linear infinite;
  ${respondTo.sm`
    width: 220px;
  `}
`;

const Title = styled.h1`
  width: 100%;
  font-size: 40px;
  text-align: center;
  padding: 30px 0;
`;

const backAndForth = keyframes`
  0% {
    left: 0;
  }
  50% {
    left: calc(100% - 2rem);
    background-color: #8660C2;
  }
  100% {
    left: 0;
  }
`;

const LogoLink = styled.div`
  width: 80px;
  margin: 0 10px;
  position: relative;
  &:before {
    content: '';
    width: 1rem;
    height: 1rem;
    background-color: #62DAFB;
    border-radius: 50%;
    position: absolute;
    left: 0;
    top: calc(50% - 0.5rem);
    transition: all 1s ease-in-out;
    animation: ${backAndForth} 1.7s infinite;
  }
  ${respondTo.xs`
    &:before {
      width: 2rem;
      height: 2rem;
      top: calc(50% - 1rem);
    }
  `}
`;

export {
    Container,
    Logo,
    Title,
    LogoLink
};
