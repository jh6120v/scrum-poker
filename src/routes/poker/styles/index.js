import styled, { css } from 'styled-components';
import { FlipCard, FlipCardFront, FlipCardBack } from './flip-card';
import { respondTo } from '../../../styles/_mixin';
import logo from '../../../assets/images/bugcat-logo-bg.png';
import logo2 from '../../../assets/images/bugcat-logo-text-bg.png';
import cardBg1 from '../../../assets/images/card-bg/card-1.gif';
import cardBg2 from '../../../assets/images/card-bg/card-2.gif';
import cardBg3 from '../../../assets/images/card-bg/card-3.gif';
import cardBg4 from '../../../assets/images/card-bg/card-4.gif';
import cardBg5 from '../../../assets/images/card-bg/card-5.gif';
import cardBg6 from '../../../assets/images/card-bg/card-6.gif';
import cardBg7 from '../../../assets/images/card-bg/card-7.gif';
import cardBg8 from '../../../assets/images/card-bg/card-8.gif';
import cardBg9 from '../../../assets/images/card-bg/card-9.gif';
import cardBg10 from '../../../assets/images/card-bg/card-10.gif';
import cardBg11 from '../../../assets/images/card-bg/card-11.gif';
import cardBg12 from '../../../assets/images/card-bg/card-12.gif';
import cardBg13 from '../../../assets/images/card-bg/card-13.gif';
import cofeeCat from '../../../assets/images/coffee-cat.png';

// flip
const PokerContainer = styled(FlipCard)`
  //height: calc(100vh - 135px);
  ${respondTo.sm`
    width: 350px;
    margin: 0 auto;
  `}
`;

const PokerListWrap = styled(FlipCardFront)`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: center;
  background: #aadff0;
  z-index: 1;
`;

const PokerSelectWrap = styled(FlipCardBack)`
  display: flex;
  justify-content: center;
  align-items: center;
  background: #aadff0;
  z-index: 1;
`;

// item auto scale. real poker radio
const PokerItemWrap = styled.div`
  display: flex;
  width: 100px;
  max-width: 22%;
  margin: 6.5px 0;
  &::after {
    content: "";
    display: block;
    padding-bottom: 140%; // 高度
  }
`;

const PokerItem = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.8rem;
  border-radius: 10px;
  background: #fff;
  cursor: pointer;
  overflow: hidden;
  position: relative;
`;

const PokerItemMask = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  transition: all 200ms linear;
  z-index: 0;
  ${(props) => {
        if (props.active) {
            return css`
              background: rgba(0,0,0,1);
            `;
        }
        return css`
          background: rgba(0,0,0,0);
        `;
    }};
`;

const PokerItemInner = styled.div`
  width: 90%;
  height: 94%;
  border-radius: 10px;
  color: #fff;
  background-color: #5bb1d4;
  display: flex;
  justify-content: center;
  align-items: center;
  text-shadow: 1px 1px 2px rgba(29, 29, 31, .7);
  background-image: url(${logo});
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
`;

const PokerSelectSafari = styled.div`
  width: 100%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  border-radius: 25px;
`;

const PokerSelectMask = styled(PokerItemMask)`
  z-index: 2;
  border-radius: 25px;
`;

const PokerSelectContainer = styled(FlipCard)`
  display: flex;
  width: 100%;
  height: auto;
  &:after {
     content: "";
     display: block;
     padding-bottom: 154%; // 高度
   }
`;

const PokerSelectedItemFront = styled(FlipCardFront)`
  display: flex;
  justify-content: center;
  border-radius: 25px;
  cursor: pointer;
  font-size: 5rem;
  color: #5AC9E8;
  background-color: #fff;
  background-size: contain;
  background-position: bottom;
  background-repeat: no-repeat;
  z-index: 1;
  &.img-1 {
    background-image: url(${cardBg1});
  }
  &.img-2 {
    background-image: url(${cardBg2});
  }
  &.img-3 {
    background-image: url(${cardBg3});
  }
  &.img-4 {
    background-image: url(${cardBg4});
  }
  &.img-5 {
    background-image: url(${cardBg5});
  }
  &.img-6 {
    background-image: url(${cardBg6});
  }
  &.img-7 {
    background-image: url(${cardBg7});
  }
  &.img-8 {
    background-image: url(${cardBg8});
  }
  &.img-9 {
    background-image: url(${cardBg9});
  }
  &.img-10 {
    background-image: url(${cardBg10});
  }
  &.img-11 {
    background-image: url(${cardBg11});
  }
  &.img-12 {
    background-image: url(${cardBg12});
  }
  &.img-13 {
    background-image: url(${cardBg13});
  }
`;

const PokerSelectedItemFrontWrap = styled.div`
  width: 100%;
  height: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  padding-top: 17%;
`;

const PokerSelectedItemFrontMask = styled(PokerItemMask)`
  z-index: 0;
  border-radius: 25px;
`;

const PokerSelectedItemBack = styled(FlipCardBack)`
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 25px;
  cursor: pointer;
  transform: rotateY(180deg);
  background-color: #fff;
  z-index: 1;
`;

const PokerSelectedItemBackWrap = styled(PokerSelectedItemFrontWrap)`
  align-items: center;
  padding-top: 0;
`;

const PokerSelectedItemBackMask = styled(PokerItemMask)`
  z-index: 2;
  border-radius: 25px;
`;

const PokerSelectedItemBackInner = styled(PokerItemInner)`
  background-image: url(${logo2});
  background-size: contain;
`;

const CloseButton = styled.div`
  z-index: 3;
  color: #fff;
  font-size: 25px;
  font-family: 'Orbitron', sans-serif;
  cursor: pointer;
  transition: all 0.2s ease-in;
  padding-left: 3px;
`;

const Setting = styled.div`
  z-index: 3;
  transition: all 0.2s ease-in;
  padding-right: 3px;
  a {
    color: #fff;
    font-size: 1.2rem;
    font-family: 'Orbitron', sans-serif;
    cursor: pointer;
  }
`;

const CoffeeCat = styled.div`
  width: ${(props) => (props.width ? props.width : '30%')};
  height: ${(props) => (props.height ? props.height : '20%')};
  background-image: url(${cofeeCat});
  background-repeat: no-repeat;
  background-size: contain;
`;

export {
    PokerContainer, PokerListWrap, PokerSelectWrap,
    PokerItemWrap, PokerItemMask, PokerItem, PokerItemInner,
    CloseButton, Setting,
    PokerSelectSafari, PokerSelectMask, PokerSelectContainer,
    PokerSelectedItemFront, PokerSelectedItemFrontWrap, PokerSelectedItemFrontMask,
    PokerSelectedItemBack, PokerSelectedItemBackWrap, PokerSelectedItemBackMask, PokerSelectedItemBackInner,
    CoffeeCat
};
