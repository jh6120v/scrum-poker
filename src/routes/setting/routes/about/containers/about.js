import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { headerTitleSet, prevLinkActGoBack } from '../../../../../modules/header';
import { SettingWrap } from '../../../styles';
import { AboutContent, AppName } from '../styles';

const About = () => {
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(headerTitleSet({
            title: 'About'
        }));

        dispatch(prevLinkActGoBack());
    }, [dispatch]);

    return (
        <SettingWrap>
            <AboutContent>
                <AppName>Capoo Cat Scrum Poker</AppName>
                This is an PWA for testing purposes only.
                <br />
                Image copyright is
                <a href="https://www.facebook.com/capoocat/" target="_new"> Capoocat </a>
                <br />
                <br />
                Author: James Yu
                <br />
                E-mail:
                <a href="mailto:jh6120v@msn.com" target="_new"> jh6120v@msn.com</a>
                <br />
                Github:
                <a href="https://github.com/jh6120v/scrum-poker" target="_new"> https://github.com/jh6120v/scrum-poker</a>
            </AboutContent>
        </SettingWrap>
    );
};

export default About;
