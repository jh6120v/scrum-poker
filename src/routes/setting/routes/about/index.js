import About from './containers/about';

export default About;