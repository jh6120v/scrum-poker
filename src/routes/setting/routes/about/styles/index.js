import styled from 'styled-components';

const AboutContent = styled.div`
  width: 100%;
  padding: 20px;
  font-family: Arial, serif;
  font-size: .8rem;
  line-height: 1.2rem;
`;

const AppName = styled.h1`
  width: 100%;
  text-align: center;
  font-size: 1.5rem;
  margin-bottom: 15px;
  font-weight: bold;
`;

export {
    AboutContent,
    AppName
};