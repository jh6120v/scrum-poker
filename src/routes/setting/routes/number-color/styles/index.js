import styled from 'styled-components';

const ColorWatch = styled.div`
  padding: 3px;
  background: #fff;
  border-radius: 1px;
  box-shadow: 0 0 0 1px rgba(0,0,0,.1);
  display: inline-block;
  cursor: pointer;
  position: relative;
`;

const ColorWatchShow = styled.div`
  width: 30px;
  height: 15px;
  border-radius: 2px;
  background-color: ${(props) => (props.color ? props.color : '#FFF')};
`;

const ColorPicker = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  margin-top: 15px;
`;

export {
    ColorWatch,
    ColorWatchShow,
    ColorPicker
};