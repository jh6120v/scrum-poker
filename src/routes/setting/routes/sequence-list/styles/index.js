import styled from 'styled-components';

const Sequence = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
`;

const Name = styled.div`
  width: 100%;
`;

const List = styled.div`
  width: 100%;
  font-size: .7rem;
  line-height: .8rem;
`;

export {
    Sequence,
    Name,
    List
};