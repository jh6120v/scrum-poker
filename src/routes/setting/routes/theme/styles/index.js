import styled from 'styled-components';
import {
    PokerItem,
    PokerItemInner,
    PokerItemWrap
} from '../../../../poker/styles';
import cardBg1 from '../../../../../assets/images/card-bg/card-1.gif';
import logo2 from '../../../../../assets/images/bugcat-logo-text-bg.png';

const ThemePreview = styled(PokerItemWrap)`
  margin: 0;
  border-radius: 10px;
  box-shadow: ${(props) => (props.selected ? '0 0 5px #8a8a8a' : 'none')};
`;

const ThemeSettingPicker = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const ThemePreviewCardListItem = styled(PokerItem)``;
const ThemePreviewCardListItemInner = styled(PokerItemInner)``;

const ThemePreviewCardFront = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  border-radius: 10px;
  color: ${(props) => (props.fontColor ? props.fontColor : '#5AC9E8')};
  background-color: ${(props) => (props.bgColor ? props.bgColor : '#ffffff')};
  background-image: url(${cardBg1});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: bottom;
`;

const ThemePreviewCardFrontInner = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  padding-top: 17%;
  font-size: 1.2rem;
`;

const ThemePreviewCardBack = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  background-color: ${(props) => (props.borderColor ? props.borderColor : '#ffffff')};
`;

const ThemePreviewCardBackInner = styled.div`
  width: 90%;
  height: 94%;
  border-radius: 10px;
  color: ${(props) => (props.fontColor ? props.fontColor : '#ffffff')};
  background-color: ${(props) => (props.bgColor ? props.bgColor : '#5bb1d4')};
  display: flex;
  justify-content: center;
  align-items: center;
  background-repeat: no-repeat;
  background-position: center center;
  background-image: url(${logo2});
  background-size: contain;
`;

const ThemeSetting = styled.div`
  display: flex;
  width: 100%;
  align-content: space-around;
  flex-wrap: wrap;
  margin-top: 15px;
`;

const ThemeSettingList = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  margin-bottom: 15px;
`;

const ThemeSettingItem = styled.div`
  position: relative;
  display: inline-block;
  border-radius: 5px;
  padding: 2px 7px;  
  margin-right: 7px;
  background-color: ${(props) => (props.color ? props.color : '#ffffff')};
  box-shadow: ${(props) => (props.selected ? '0 0 3px #8a8a8a' : 'none')};
  &:before {
    content: attr(data-title);
    mix-blend-mode: difference;
  }
  &:last-child {
    margin-right: 0;
  }
`;

const ColorWatch = styled.div`
  padding: 3px;
  background: #fff;
  border-radius: 1px;
  box-shadow: 0 0 0 1px rgba(0,0,0,.1);
  display: inline-block;
  cursor: pointer;
  position: relative;
`;

const ColorWatchShow = styled.div`
  width: 30px;
  height: 15px;
  border-radius: 2px;
  background-color: ${(props) => (props.color ? props.color : '#FFF')};
`;

const ColorPicker = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  margin-top: 15px;
`;

export {
    ThemePreview,
    ThemeSetting,
    ThemeSettingList,
    ThemeSettingItem,
    ThemeSettingPicker,
    ThemePreviewCardListItem,
    ThemePreviewCardListItemInner,
    ThemePreviewCardFront,
    ThemePreviewCardFrontInner,
    ThemePreviewCardBack,
    ThemePreviewCardBackInner,
    ColorWatch,
    ColorWatchShow,
    ColorPicker
};
