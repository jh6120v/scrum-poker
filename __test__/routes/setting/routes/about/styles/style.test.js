import 'jest-styled-components';
import { render } from '@testing-library/react';
import React from 'react';
import { AboutContent, AppName } from '../../../../../../src/routes/setting/routes/about/styles';

describe('test about style.', () => {
    it('test <AboutContent /> snapshot.', () => {
        const { container } = render(<AboutContent />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <AboutContent /> when render it.', () => {
        const { container } = render(<AboutContent />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('padding', '20px');
        expect(tree).toHaveStyleRule('font-family', 'Arial,serif');
        expect(tree).toHaveStyleRule('font-size', '.8rem');
        expect(tree).toHaveStyleRule('line-height', '1.2rem');
    });

    it('test <AppName /> snapshot.', () => {
        const { container } = render(<AppName />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <AppName /> when render it.', () => {
        const { container } = render(<AppName />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('text-align', 'center');
        expect(tree).toHaveStyleRule('font-size', '1.5rem');
        expect(tree).toHaveStyleRule('margin-bottom', '15px');
        expect(tree).toHaveStyleRule('font-weight', 'bold');
    });
});
