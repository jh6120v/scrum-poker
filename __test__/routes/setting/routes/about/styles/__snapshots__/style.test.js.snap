// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test about style. test <AboutContent /> snapshot. 1`] = `
.c0 {
  width: 100%;
  padding: 20px;
  font-family: Arial,serif;
  font-size: .8rem;
  line-height: 1.2rem;
}

<div
  class="c0"
/>
`;

exports[`test about style. test <AppName /> snapshot. 1`] = `
.c0 {
  width: 100%;
  text-align: center;
  font-size: 1.5rem;
  margin-bottom: 15px;
  font-weight: bold;
}

<h1
  class="c0"
/>
`;
